tiny_obj_loader.h

```diff
@@ -219,19 +219,42 @@ typedef struct {
   int texcoord_index;
 } index_t;
 
-typedef struct {
+// mesh_t will contain faces corresponsing to a smoothing group id.
+struct mesh_t{
+  int smoothingGroupId;                          // Smoothing group id. 
   std::vector<index_t> indices;
   std::vector<unsigned char> num_face_vertices;  // The number of vertices per
                                                  // face. 3 = polygon, 4 = quad,
                                                  // ... Up to 255.
   std::vector<int> material_ids;                 // per-face material ID
   std::vector<tag_t> tags;                       // SubD tag
-} mesh_t;
 
-typedef struct {
+  // Constructor
+  mesh_t(int smGrpId = 0) { smoothingGroupId = smGrpId; }
+};
+
+// shape_t will contain all faces under a group.
+// The faces will be stored in a collection of mesh_t, 
+// each mesh_t corresponds to a smoothing group id.
+struct shape_t{
   std::string name;
-  mesh_t mesh;
-} shape_t;
+  std::vector<mesh_t> meshes;
```

Comment:
New proposal: A shape_t will contain a collection (std::vector) of mesh_t. Each mech_t corresponds to one smoothigngrup.

I think, this representation confirms to the wavefront .obj file specification. 

This also requires minimum change to the existing code flow.